pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -109,13 +108,15 @@ def new_process(self, process_idx: int, trainer, mp_queue) -> None:
 
         # replace trainer save_checkpoint to use `xm.save`
         trainer.save_checkpoint = self.save_checkpoint
-        self.barrier()
+        self.barrier("pre-run-stage")
 
         results = trainer.run_stage()
 
         self.__save_end_of_training_weights(self.lightning_module)
         self.transfer_distrib_spawn_state_on_fit_end(results)
 
+        self.barrier("end-process")
```

Comment:
is this needed?

Reply:
Oh yes! We need a `rendezvous` before the end of the process.